src/praisonai-agents/praisonaiagents/skills/manager.py

```
@@ -126,15 +137,29 @@ def activate_by_name(self, name: str) -> bool:
+        for skill in self._skills.values():
+            # Skip if explicitly disabled for model invocation
+            if getattr(skill.properties, "disable_model_invocation", False):
+                continue
+                
+            # Check capability requirements if enforcement is strict
+            if self._validator.enforcement_level == EnforcementLevel.STRICT:
+                try:
+                    validation = self.validate_skill_capabilities(skill.properties.name)
+                    if validation.state == SkillState.UNAVAILABLE:
+                        continue  # Skip unavailable skills in strict mode
+                except Exception as e:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)